src/analysis/simulation.jl

```diff
@@ -1,35 +1,35 @@
 
 """
 Perform EPANET hydraulic simulation (via WNTR) using controls in WM
-solution. 
+wm_solution. 
 """
-function simulate(data::Dict{String,Any}, solution::Dict{String,Any},
+function simulate(wm_data::Dict{String,Any}, wm_solution::Dict{String,Any},
                      inpfilepath::String)
 
     # populate a wntr network with simulation results
     wn = wntr.network.model.WaterNetworkModel(inpfilepath)
 
     # store pump names in a set
     pump_set = Set()
-    for (key,pump) in solution["1"]["pump"]
-        pump_name = data["pump"][key]["name"] # epanet name
+    for (key,pump) in wm_solution["solution"]["nw"]["1"]["pump"]
+        pump_name = wm_data["nw"]["1"]["pump"][key]["name"] # epanet name
         push!(pump_set, pump_name)
     end
```

Comment:
You could replace this with something like `pump_set = [pump["name"] for (i, pump) in wm_data["nw"]["1"]["pump"]]`.

Reply:
Looks like Yang made this change